akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazySinkSpec.scala

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import java.util.concurrent.TimeoutException
+
+import akka.stream.ActorAttributes.supervisionStrategy
+import akka.stream.Supervision.resumingDecider
+import akka.stream._
+import akka.stream.testkit.TestPublisher
+import akka.stream.testkit.TestSubscriber.Probe
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.testkit.AkkaSpec
+
+import scala.concurrent.{ Promise, Future, Await }
+import scala.concurrent.duration._
+
+class LazySinkSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+    .withInputBuffer(initialSize = 1, maxSize = 1)
+  implicit val materializer = ActorMaterializer(settings)
+
+  val ex = TE("")
+
+  "A LazySink" must {
+    "work in happy case" in assertAllStagesStopped {
+      val futureProbe = Source(0 to 10).runWith(Sink.lazySink[Int, Probe[Int]](_ ⇒ Future.successful(TestSink.probe[Int]), () ⇒ ???))
```

Comment:
to check that we do not call this metho - can set this to some default value otherwise.